cpp/README.md

```diff
@@ -0,0 +1,62 @@
+# Lance: A Columnar File Format
+
+
+# Development
+
+## Code Style
+
+The core of `Lance` data format is developed in `C++20`.
+
+The code base follows Apache Arrow and [Google C++ Code Style](https://google.github.io/styleguide/cppguide.html).
+With exceptions:
+* Line width: 100
+
+The code style is enforced via [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html). Please make sure `clang-format` has been set up appropriately in the IDE.
+
+## Build
+
+The code is developed on `macOS 12+` and a recent Linux (`Ubuntu 22.04`). Please file bugs or contribute fixes if you encounter issues with other Linux flavors.
+
+A recent C++ compiler that supports C++20 is needed. The toolchain that has been tested is:
+
+* AppleClang >= 13.1.6
+* GCC >= 11.2
+* CMake >= 3.22
+* Apache Arrow >= 8.0
+* Protobuf > 3.12
+
+```sh
+# On macOS 12+
+brew install apache-arrow cmake protobuf
+# Optionally, build document
+brew install doxygen
+```
+
+```sh
+# On Ubuntu 22.04
+# Follow https://arrow.apache.org/install/
+sudo apt update
```

Comment:
I would say just link to arrow is fine. If they update their instructions there's no need to have to update ours too

Reply:
done